libvast/builtins/formats/zeek_tsv.cpp

```diff
@@ -0,0 +1,757 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/arrow_table_slice.hpp"
+#include "vast/cast.hpp"
+#include "vast/concept/parseable/string/any.hpp"
+#include "vast/concept/parseable/vast/pipeline.hpp"
+#include "vast/concept/printable/to_string.hpp"
+#include "vast/concept/printable/vast/json.hpp"
+#include "vast/data.hpp"
+#include "vast/detail/string.hpp"
+#include "vast/detail/string_literal.hpp"
+#include "vast/detail/to_xsv_sep.hpp"
+#include "vast/detail/zeekify.hpp"
+#include "vast/generator.hpp"
+#include "vast/plugin.hpp"
+#include "vast/table_slice_builder.hpp"
+#include "vast/to_lines.hpp"
+#include "vast/type.hpp"
+#include "vast/view.hpp"
+
+#include <arrow/record_batch.h>
+#include <caf/error.hpp>
+#include <caf/expected.hpp>
+#include <caf/none.hpp>
+#include <fmt/core.h>
+#include <fmt/format.h>
+
+#include <algorithm>
+#include <cctype>
+#include <iterator>
+#include <string>
+#include <string_view>
+
+namespace vast::plugins::zeek_tsv {
+
+namespace {
+// The type name prefix to preprend to zeek log names when transleting them
+// into VAST types.
+constexpr std::string_view type_name_prefix = "zeek.";
+constexpr auto default_sep = '\x09';
+constexpr auto default_set_sep = ',';
+constexpr std::string_view default_empty_val = "(empty)";
+constexpr std::string_view default_unset_val = "-";
+
+/// Constructs a polymorphic Zeek data parser.
+template <class Iterator, class Attribute>
+struct zeek_parser_factory {
+  using result_type = rule<Iterator, Attribute>;
+
+  zeek_parser_factory(const std::string& set_sep) : set_sep_{set_sep} {
```

Comment:
```suggestion
  explicit zeek_parser_factory(const std::string& set_sep) : set_sep_{set_sep} {
```

Reply:
I added the `explicit` qualifier in https://github.com/tenzir/vast/pull/3114/commits/6235858de3930fb5890c74f464960c5e503a5f6a